straxen/plugins/veto_events.py

```diff
@@ -0,0 +1,126 @@
+import strax
+import straxen
+
+import numpy as np
+import numba
+from immutabledict import immutabledict
+
+
+@strax.takes_config(
+    strax.Option('event_left_extension_nv', default=0,
+                 help="Extends events this many ns to the left"),
+    strax.Option('event_resolving_time_nv', default=300,
+                 help="Resolving time for fixed window concidene [ns]."),
+    strax.Option('event_min_hits_nv', default=6,
+                 help="Minimum number of fully convinet hitlets to define an event."),
+    strax.Option('gain_model_nv',
+                 help='PMT gain model. Specify as (model_type, model_config)'),
+    strax.Option('n_nveto_pmts', type=int,
+                 help='Number of TPC PMTs'),
+    strax.Option('channel_map', track=False, type=immutabledict,
+                 help="immutabledict mapping subdetector to (min, max) "
+                      "channel number."),
+)
+class nVETOEvents(strax.OverlapWindowPlugin):
```

Comment:
Do you forsee a similar plugin for MV, and if so, should we generalize some of the naming?

Reply:
Yes, the naming should already be such, that we can directly make the corresponding child plugin. But, I need to discuss this with Andrea first.